screens/CreateGame.js

```diff
@@ -0,0 +1,257 @@
+import TextInput from "../common/TextInput";
+import Container from "../common/Container";
+import RNPickerSelect from "react-native-picker-select";
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import { DataStore, Auth } from "aws-amplify";
+import {Player, Game, Location, GamePlayer, Rsvp} from "../src/models";
+import ErrorPopup from "../common/ErrorPopup";
+import '@azure/core-asynciterator-polyfill';
+
+function CreateGame({ navigation }) {
+
+    const [loading, setLoading] = useState(false);
+    const [ gameName, setGameName] = useState("");
+    const [ gameDescription, setGameDescription] = useState("");
+    const [ gameLocation, setLocation] = useState("");
+    const [ gameSkillLevel, setSkillLevel] = useState("");
+    const [chosenDate, setChosenDate] = useState(new Date());
+    const [showError, setShowError] = useState(false);
+    const [errorMessage, setErrorMessage] = useState("");
+
+//on change function to set user selected date 
+const changeSelectedDate = (event, selectedDate) => {
+    const currentDate = selectedDate || chosenDate;
+    setChosenDate(currentDate);
+}
+
+//helper function to get everyone but the organizer to test inviting players
+async function getAllPlayers() {
+  let userEmail = "Test@gmail.com";
+  const allPlayers = await DataStore.query(Player, (p) => p.email.ne(userEmail) );
+  return allPlayers;
+}
+
+//gets current user
+async function getPlayer(){
+    response = await Auth.currentUserInfo();
+    const player = await DataStore.query(Player, (p) => p.email.eq(response.attributes.email));
+    return player[0];
+}
+//gets the location for the game based on what the user specified 
+async function getLocation(locationName) {
+  const location = await DataStore.query(Location, (l) => l.name.eq(locationName));
+  return location;
+}
+
+//saves a name game or returns an error
+async function storeGame(gameName, gameDescription, epochDate, gameSkillLevel, organizer, location) {
+
+    //form field validation
+    // if game name is blank default name is generated
+    if([gameName].includes("")){
+        gameName = "Pickup game at " + location[0].name;
+    }
+    //if no game description is provided then set empty string
+    if([gameDescription].includes("")){
+        gameDescription = "";
+    }
+    //if no location is selected display error message
+    if([gameLocation].includes("")){
+        setShowError(true);
+        setErrorMessage("Please select a location");
+        return;
+    }
+    //if gameskilllevel is not specified set to ANY as default
+    if([gameSkillLevel].includes("")){
+        gameSkillLevel = "ANY";
+    }
+    //check if specified datetime is in the past 
+    epochNow = Math.floor(Date.now() / 1000);
+    if(epochDate <= epochNow){
+        setShowError(true);
+        setErrorMessage("Please select a valid future date and time");
+        return;
+    }
+    setShowError(false);
+    navigation.navigate("HomeScreen")
```

Comment:
This line should be last in the function. We want the API call to finish before we navigate to a new screen. I also don't know how this effects the flow of execution if we have code that we want to execute AFTER we navigate. Also, after the last API call you have to make, make sure to setLoading(false).